### PR TITLE
Added difficulty levels according to coin density

### PR DESCRIPTION
**Pull request by kartikct25**
@vineetjc  I have added difficulty level modes that changes crowdness of coins

### CODE REVIEW

File: src/managers/game_manager.py [hunk 1]
```diff
@@ -7,8 +7,11 @@ class Game_manager(object):
     def __init__(self):
         self.score = 0
         self.time = 0
-
+	self.Difficulty = {"Easy":25,"Medium":15,"Hard":5}
```
Comment (amrit-choudhary) [inline src/managers/game_manager.py]: use
self.difficulty
please follow the naming convention

File: src/game_screens/gameMode_screen.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+from src.game_screens.screen import Screen
```
Comment (amrit-choudhary) [inline src/game_screens/gameMode_screen.py]: rename file and class to game_mode_screen

Comment (amrit-choudhary) [reply src/game_screens/gameMode_screen.py]: class to
Game_Mode_Screen

**Comment by amrit-choudhary** [comment]
The way you are passing difficulty from game_mode screen to game_screen is bad.

1. Create a new Enum in game_enums called Difficulty, give it 3 values
```
class Difficulty(IntEnum):
    EASY = 25
    MEDIUM = 15     
    HARD = 5
````
2. In the Game_Manager, create a new var called difficulty, initially set to Difficulty.MEDIUM
3. main_py creates a Game_Manager object, every one should use that. The game will always have a single Game_Manager.
4. In main.py, in the init function of game_mode_screen, give it the game_manager var. It will save the game_manager for later use. Check how game_over_screen does it.
5. In game_mode_screen, when you click on, let's say, the Easy button, it will set 
````
self.game_manager.difficulty = Difficulty.Easy
return GameMode.Game
````
that's it, nothing more.
6. Get difficulty value by
````
self.difficulty = int(self.game_manager.difficulty)
````
this will either give you 25, 15 or 5 based on difficulty.
Nothing more is needed. Use this difficulty value in your calculations.

Please try this and tell me when there is a problem.

**Comment by kartikct25** [comment]
sure

**Comment by amrit-choudhary** [comment]
also, if you want to start afresh
you can use revert commit and start from scratch

**Comment by kartikct25** [comment]
I have made all changes you said....

And also changed Game_Over_Screen to return GAME_MODE  screen when we press restart

**Comment by kartikct25** [comment]
Do I again have to create a PR

**Comment by kartikct25** [comment]
I Don't have to ,,right?

**Comment by amrit-choudhary** [comment]
yeah

File: src/game_screens/game_screen.py [hunk 3]
```diff
@@ -9,23 +9,25 @@
...
+    def __init__(self, pygame, res, surface, size, gameclock, game_manager):    
         Screen.__init__(self, pygame, res, surface)
-
+           
         # set up initial variables
         self.need_reset = False
         self.size = size
-        self.result = 0
-        self.i = 1
+	self.result = 0
```
Comment (amrit-choudhary) [inline src/game_screens/game_screen.py]: indentation problems like this

**Comment by amrit-choudhary** [comment]
change this
``` python
self.game_manager.score = self.cart.points
````
to
``` python
self.game_manager.score = self.cart.points * some_difficulty_based_multiplier
```

**Comment by amrit-choudhary** [comment]
there are other indentation problems too, with other filed that you have changed.

File: src/game_screens/game_screen.py (hunk 3 above)
Comment (kartikct25) [reply src/game_screens/game_screen.py]: @amrit-choudhary  Everything is indented properly in my code but it shows these errors on github ..
Can you help me?

Comment (amrit-choudhary) [reply src/game_screens/game_screen.py]: can you show a screenshot from your editor of the same code?

Comment (kartikct25) [reply src/game_screens/game_screen.py]: image

Comment (kartikct25) [reply src/game_screens/game_screen.py]: is it related to push ??

Comment (amrit-choudhary) [reply src/game_screens/game_screen.py]: I don't know.
I think it's related to the editor.
[link](https://stackoverflow.com/questions/29074158/indentation-inconsistency-python-taberror-different-editors)
I use VS Code.

**Comment by kartikct25** [comment]
I will update it tomorrow evening ..because i will be in train

**Comment by kartikct25** [comment]
I have fixed all indentations...

And also I applied score multiplier but it increases score for higher difficulty rapidly 
Which I don't think we should use in difficulty level ...Current mode is good enough for leveling score 

We can use it in infinite mode where score multiplier will increase according to time we survive

Btw I'll update my corrected code tomorrow evening

**Comment by vineetjc** [comment]
A lot of conversation haha :p Are we waiting for updates? @amrit-choudhary @kartikct25

**Comment by vineetjc** [comment]
Some very interesting observations right on my first try at this:

- Although I like the idea, I scored more on HARD (a whopping 100+ points) than on EASY. I think 'hardness' of the game is much more than the density/crowdedness of the things that fall, maybe it's about changing how I initially randomized the falling of objects.
- The high score should be updated wrt difficulty, it shouldn't be one all across the modes.

Other things:

- Not sure if your repository isn't updated, please do so, some bugs were fixed but still persistent on this PR.
- Also not sure who put that wolf background, it looks good(!), but does it suit the game? Need your opinions.

Let me know if I forgot anything else, and I think the above conversation has points I don't need to repeat myself.

File: src/game_screens/game_screen.py (hunk 3 above)
Comment (vineetjc) [reply src/game_screens/game_screen.py]: @kartikct25 I use Atom, but that's beside the point I'm going to make: there should be options in most of the latest editors (like Atom, which is why I gave my example) that deal with whitespaces and other formatting things etc. That should save you from manually checking it out

**Comment by amrit-choudhary** [comment]
@vineetjc 
- defining hardness is a continuous process, we can add/remove a lot of things that define hardness. 
He has implemented one parameter, I suggested another one, coin fall speed. We should have a discussion, agree on some parameters and add them one by one.
- that's an option too. like my last point, decisions like these are game design decisions, not programming ones.
- I like the current bg, not the wolf one.

**Comment by amrit-choudhary** [comment]
@vineetjc @kartikct25 
Only some indentation errors are left I think. If you can, you should merge this, fix the small errors.
My final version of #37 is waiting for this to get merged.

File: src/game_screens/game_mode_screen.py [hunk 4]
```diff
@@ -26,16 +26,16 @@ def update(self, events):
...
-		self.game_manager.difficulty = Difficulty.EASY
-		return Game_mode.GAME
-            
+                self.game_manager.difficulty = Difficulty.EASY
+                return Game_mode.GAME
+
             if self.buttons['Medium'].check_click(mouseup_event.pos):
-		self.game_manager.difficulty = Difficulty.MEDIUM
-		return Game_mode.GAME
+	        self.game_manager.difficulty = Difficulty.MEDIUM
```
Comment (amrit-choudhary) [inline src/game_screens/game_mode_screen.py]: this is wrong, i guess

File: src/misc/game_enums.py [hunk 5]
```diff
@@ -23,7 +23,7 @@ class Entity(Enum):
 
 class Difficulty(Enum):
     ''' Enum to provide difficulty level. '''
-    EASY = 25
-    MEDIUM = 15
-    HARD = 5 
+    EASY = {"DENSITY":25,"SCORE_MULTIPLIER":1}
+    MEDIUM = {"DENSITY":15,"SCORE_MULTIPLIER":3}
+    HARD = {"DENSITY":5,"SCORE_MULTIPLIER":5}
```
Comment (amrit-choudhary) [inline src/misc/game_enums.py]: Wow this is awesome

**Comment by amrit-choudhary** [comment]
This PR is completely fine, i'm just curious

**Comment by kartikct25** [comment]
> This PR is completely fine, i'm just curious

I can add it again it was removed because of some  network  issue i had to again edit previous one

So shall I add it again??

**Comment by kartikct25** [comment]
i am adding it and increase score according to difficulty

**Comment by amrit-choudhary** [comment]
the system is nice.
you should check if it's working before commiting.
i think there should be some change in game_screen if you want to use that.

**Comment by amrit-choudhary** [comment]
if this works
EASY = {"DENSITY":25,"SCORE_MULTIPLIER":1}
this is a lot better than the old system of
EASY = 25

**Comment by kartikct25** [comment]
i applied it and it's running...check it

**Comment by vineetjc** [comment]
@kartikct25 Merged this due to significant changes that are needed later. I'll put my suggestions elsewhere again, maybe as an issue or something because it's a game design decision as said by @amrit-choudhary

**Pull request merged**